frame/support/procedural/src/construct_runtime/expand/config.rs

```diff
@@ -97,6 +99,12 @@ pub fn expand_outer_config(
 				Ok(())
 			}
 		}
+
+		impl #scrate::traits::RuntimeGenesisBuild for RuntimeGenesisConfig {
+			fn build(&self) {
+				#genesis_build_calls
```

Comment:
This is missing the call to `on_genesis`.

Reply:
added in https://github.com/paritytech/substrate/pull/14306/commits/87d969ae6bf68b468dd0c7ecbdb081db67fb86ed